### PR TITLE
Circuitpython nickzoic 1045 nrf rotaryio

### PR DESCRIPTION
**Pull request by nickzoic**
Okay this is an implementation for adafruit/circuitpython#1045 very closely modeled on the atmel-samd port implementation by Scott.

### CODE REVIEW

File: ports/nrf/common-hal/rotaryio/IncrementalEncoder.c [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+    self->pin_a = pin_a->number;
+    self->pin_b = pin_b->number;
+
+    _objs[self->pin_a] = self;
+    _objs[self->pin_b] = self;
+
+    nrfx_gpiote_in_config_t cfg = {
+        .sense = NRF_GPIOTE_POLARITY_TOGGLE,
+        .pull = NRF_GPIO_PIN_NOPULL,
```
Comment (tannewt) [inline ports/nrf/common-hal/rotaryio/IncrementalEncoder.c]: I think you want pull up. As is, without external pull-ups, the output values are very sporadic.

**Comment by nickzoic** [comment]
Ah, no problem. The hardware I'm using has internal pull up resistors so it works great, but lots don't so it's probably a sensible default.

**Comment by tannewt** [comment]
@nickzoic Let me know when you've added the pull up setting.

**Comment by nickzoic** [comment]
OK @tannewt that's done and works nicely on my hardware with the power wire yanked, whereas it didn't work at all with no pullup power.
(I'm using a little board with a similar encoder to the adafruit one, plus two 10K pullup resistors.)

**Comment by Tasm-Devil** [comment]
@nickzoic Thx for the explanation. It very cool bit manipulation. I like this kinda stuff. :+1:

**Pull request merged**